{
  "cveId": "CVE-2024-28179",
  "title": "Jupyter Server Proxy's Websocket Proxying does not require authentication",
  "description": "Jupyter Server Proxy allows users to run arbitrary external processes alongside their Jupyter notebook servers and provides authenticated web access. Prior to versions 3.2.3 and 4.1.1, Jupyter Server Proxy did not check user authentication appropriately when proxying websockets, allowing unauthenticated access to anyone who had network access to the Jupyter server endpoint. This vulnerability can allow unauthenticated remote access to any websocket endpoint set up to be accessible via Jupyter Server Proxy. In many cases, this leads to remote unauthenticated arbitrary code execution, due to how affected instances use websockets. The websocket endpoints exposed by `jupyter_server` itself is not affected. Projects that do not rely on websockets are also not affected. Versions 3.2.3 and 4.1.1 contain a fix for this issue.",
  "publishedDate": "2024-03-20T19:54:38.247000+00:00",
  "lastModifiedDate": "2024-08-02T00:48:49.393000+00:00",
  "severity": "CRITICAL",
  "cvssMetrics": [
    {
      "version": "3.1",
      "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:C/C:H/I:H/A:H",
      "baseScore": 9.1,
      "baseSeverity": "CRITICAL"
    }
  ],
  "epss": {
    "score": 0.0026,
    "percentile": 49.169
  },
  "riskScore": 44,
  "exploitationStatus": "UNKNOWN",
  "affectedSystems": {
    "cpeMatches": [],
    "vendors": ["jupyterhub"],
    "products": ["jupyter-server-proxy"]
  },
  "references": [
    {
      "url": "https://github.com/jupyterhub/jupyter-server-proxy/security/advisories/GHSA-w3vc-fx9p-wp4v",
      "source": "https://github.com/jupyterhub/jupyter-server-proxy/security/advisories/GHSA-w3vc-fx9p-wp4v",
      "tags": ["x_refsource_CONFIRM"]
    },
    {
      "url": "https://github.com/jupyterhub/jupyter-server-proxy/commit/764e499f61a87641916a7a427d4c4b1ac3f321a9",
      "source": "https://github.com/jupyterhub/jupyter-server-proxy/commit/764e499f61a87641916a7a427d4c4b1ac3f321a9",
      "tags": ["x_refsource_MISC"]
    },
    {
      "url": "https://github.com/jupyterhub/jupyter-server-proxy/commit/bead903b7c0354b6efd8b4cde94b89afab653e03",
      "source": "https://github.com/jupyterhub/jupyter-server-proxy/commit/bead903b7c0354b6efd8b4cde94b89afab653e03",
      "tags": ["x_refsource_MISC"]
    },
    {
      "url": "https://github.com/jupyterhub/jupyter-server-proxy/blob/9b624c4d9507176334b46a85d94a4aa3bcd29bed/jupyter_server_proxy/handlers.py#L433",
      "source": "https://github.com/jupyterhub/jupyter-server-proxy/blob/9b624c4d9507176334b46a85d94a4aa3bcd29bed/jupyter_server_proxy/handlers.py#L433",
      "tags": ["x_refsource_MISC"]
    }
  ],
  "attackTechniques": [],
  "metadata": {
    "tags": ["CWE-306"],
    "requiresUserInteraction": null,
    "requiresPrivileges": null,
    "attackVector": null,
    "sources": [
      {
        "name": "CVEList",
        "url": "https://github.com/CVEProject/cvelistV5/blob/main/cves/2024/28xxx/CVE-2024-28179.json",
        "lastModified": "2024-08-02T00:48:49.393000+00:00"
      }
    ]
  }
}
